Assets/Scripts/Game Object/Card/Card.cs

```diff
@@ -0,0 +1,64 @@
+using UnityEngine;
+using UnityEngine.UI;
+using UnityEngine.EventSystems;
+
+public abstract class Card : MonoBehaviour, IDragHandler, IBeginDragHandler, IEndDragHandler
+{
+    [SerializeField]
+    protected float cooldown = 15;
+    [SerializeField]
+    protected PlayerController player;
+    protected Image cardImage;
+    protected Button cardButton;
+    protected Vector2 cardOriginPos;
+    protected RectTransform rectTransform;
```

Comment:
Some of these variables can be made private, but it doesn't matter that much. `TryPlayCard` can also be made protected.